apps_download/models/product_product.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+import os
+import tempfile
+import shutil
+import logging
+import base64
+import subprocess
+from odoo.exceptions import ValidationError
+from odoo import models, fields, api, _
+_logger = logging.getLogger(__name__)
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    dependent_product_ids = fields.Many2many(
+        'product.product', 'prto_validateoduct_dependent_rel',
```

Comment:
The name of relationship could be improved

Reply:
Sorry, I do not get you. do you mean Table name?

Reply:
Why is the relationship being manually defined? Odoo does this by default, and uses a very specific naming schema that is easy to identify. You shouldn't need to manually define unless the relationship name is above the Postgres size limit, or you are trying to use one that already exists somewhere else.

Reply:
@lasley I have added the relationship manually because my relation is from product table to product table itself, and IMO better to maintain our one schema specifically when you are using both models from the odoo core to create the relationship, Else if someone has some customization with default schema then it is going to use the same Table which can cause the problem.
IMO the current name is not good enough to interpret the meaning I think I should have to change the current table name ,
Let me know your view for the same.

Reply:
stlll "prto_validateoduct" looks like the cursor in the editor moved while you were typing...